test/dynamo/test_subclasses.py

```diff
@@ -2471,17 +2471,31 @@ def forward(self, primals_1: "Sym(s2)", primals_2: "Sym(s3)", primals_3: "Sym(s1
         clone: "f64[s0, s1]" = torch.ops.aten.clone.default(primals_4);  primals_4 = None
 
         mul: "f64[s0, s1]" = torch.ops.aten.mul.Tensor(clone, primals_1);  clone = None
-        return (mul, primals_5, primals_6, primals_7, primals_8, primals_10, primals_10, primals_1, primals_8, primals_10)
+        return (mul, primals_5, primals_6, primals_7, primals_8, primals_10, primals_10, primals_5, primals_1, primals_8, primals_10)
 """,  # noqa: B950
         )
 
         self.assertExpectedInline(
             normalize_gm(bw[0].print_readable(print_output=False)),
             """\
 class GraphModule(torch.nn.Module):
-    def forward(self, primals_1: "Sym(s2)", primals_8: "Sym(s2)", primals_10: "Sym(s1)", tangents_1: "f64[s0, s1]", tangents_2: "i64[s2 + 1]", tangents_3: "f32[s5, 0]", tangents_4: "f32[s6, 0]"):
+    def forward(self, primals_1: "Sym(s2)", primals_8: "Sym(s2)", primals_10: "Sym(s1)", primals_5: "i64[s2 + 1]", tangents_1: "f64[s0, s1]", tangents_2: "i64[s2 + 1]", tangents_3: "f32[s5, 0]", tangents_4: "f32[s6, 0]", tangents_token: "f32[0]"):
+        with_effects = torch.ops.higher_order.with_effects(tangents_token, torch.ops.nested._assert_equal.default, primals_5, tangents_2, 'invalid gradient at index 0 - got [s2, s3, s1] but expected shape compatible with [s2, s3, s1]');  tangents_token = None
+        getitem: "f32[0]" = with_effects[0];  with_effects = None
+        with_effects_1 = torch.ops.higher_order.with_effects(getitem, torch.ops.nested._assert_equal.default, tangents_2, primals_5, 'sum_to_size(): sizes unsupported');  getitem = None
+        getitem_2: "f32[0]" = with_effects_1[0];  with_effects_1 = None
+
         mul_1: "f64[s0, s1]" = torch.ops.aten.mul.Tensor(tangents_1, primals_1);  tangents_1 = primals_1 = None
-        return (None, None, None, mul_1, tangents_2, tangents_3, tangents_4, primals_8, primals_10, primals_10)
+        with_effects_2 = torch.ops.higher_order.with_effects(getitem_2, torch.ops.nested._assert_equal.default, primals_5, tangents_2, 'invalid gradient at index 0 - got [s2, s3, s1] but expected shape compatible with [s2, s3, s1]');  getitem_2 = None
+        getitem_4: "f32[0]" = with_effects_2[0];  with_effects_2 = None
+        with_effects_3 = torch.ops.higher_order.with_effects(getitem_4, torch.ops.nested._assert_equal.default, tangents_2, primals_5, 'sum_to_size(): sizes unsupported');  getitem_4 = None
```

Comment:
Not sure we can do about these in eager without union find, but maybe we can do some kind of pass during compile (after partitioner?) to get rid of duplicate checks in the graph